app/models/organisation_relationship.rb

```diff
@@ -1,4 +1,28 @@
 class OrganisationRelationship < ApplicationRecord
   belongs_to :child_organisation, class_name: "Organisation"
   belongs_to :parent_organisation, class_name: "Organisation"
+  validates :parent_organisation_id, uniqueness: { scope: [:child_organisation_id] }
+  validate :validate_housing_provider_relationship, on: :housing_provider
+  validate :validate_managing_agent_relationship, on: :managing_agent
+
+private
+
+  def validate_housing_provider_relationship
+    if parent_organisation_id.blank?
+      child_organisation.errors.add :related_organisation_id, "You must choose a housing provider"
+    elsif OrganisationRelationship.exists?(child_organisation:, parent_organisation:)
+      child_organisation.errors.add :related_organisation_id, "You have already added this housing provider"
+    elsif !parent_organisation.holds_own_stock
+      errors.add :parent_organisation_id
```

Comment:
I had to add an error to the org relationship model as well here so the relationship wasn't incorrectly saved, as the displayed error is being added to the org so doesn't technically invalidate the relationship.

Reply:
done by using an org relationship model in the view rather than an org